src/http/client.cr

```diff
@@ -764,7 +764,7 @@ class HTTP::Client
     socket = @socket
     return socket if socket
 
-    hostname = @host.starts_with?('[') && @host.ends_with?(']') ? @host[1..-2] : @host
+    hostname = host.lchop?('[').try &.rchop?(']') || host
```

Comment:
This change doesn't optimize. On the contrary, it allocates an additional intermediary string for bracketed hostname.

It also accepts an hostname only starting with a bracket but not ending with it. I suppose this is considered invalid.

Reply:
I doesn't accept a hostname only starting with a bracket, because it's `lchop?`.
If not starting with `[`, it will return nil.

Not sure it's less optimized, because there is no `Char::Reader` involved.
I have to benchmark it

Reply:
Please don't invest to much time in such tiny optimizations which might have an effect or might not. And yes, allocating a string is likely less efficient than a char reader. The reader could be avoided using `byte_slice`, though.
In any case, it's not related to this PR.

Reply:
Now worry, hat's because I already used this logic already in my code to fix this issue

Reply:
Now worry, that's because I already used this logic already in my code to fix this issue

Reply:
No worry, that's because I already used this logic already in my code to fix this issue